### PR TITLE
docs: vue-router composition api example

### PR DESCRIPTION
**Pull request by marina-mosti**
Adds examples on how to mock vue-router 4 when using `useRouter` and `useRoute` on a setup function through composition API.

### CODE REVIEW

**Comment by cexbrayat** [comment]
Hi @marina-mosti 
Thx for PR!

I had that on my todo list for a long time. I was even considering mentioning https://github.com/posva/vue-router-mock which makes it even easier. What do you think?

**Comment by marina-mosti** [comment]
Hey @cexbrayat this lib is really nice 👍 from the educational point of view though explaining how to do it without third party libraries is what I would lean to though, since it's can also be applied to mocking other things

**Review by cexbrayat — APPROVED** [review]

**Comment by cexbrayat** [comment]
Looks good. I'll let @afontcu give it a look when he has time.

Maybe we could just add a mention at the end to vue-router-mock? I think not enough people know that Eduardo maintains this tiny lib to make router tests even easier to write.

File: docs/guide/advanced/vue-router.md [hunk 1]
```diff
@@ -301,6 +301,103 @@ test('routing', async () => {
...
+}
+```
+
+This time in order to test the component, we will use jest's ability to mock an imported resource, `vue-router` and mock both the router and route directly.
+
+```js
+import { useRouter, useRoute } from 'vue-router'
+
+jest.mock('vue-router', () => ({
+  useRoute: jest.fn(() => ({ })),
```
Comment (lmiller1990) [inline docs/guide/advanced/vue-router.md:341]: This could just be `jest.fn()`?

**Comment by lmiller1990** [comment]
Definitely agree we could promote vue-router-mock, but thanks for documenting this.

**Review by afontcu — APPROVED** [review]
> Maybe we could just add a mention at the end to vue-router-mock?

+1 to that, even though this PR looks fine (other than the little comment Lachlan added) 😄  we can always add a mention to eduardo's library in a future iteration.

thanks for this!

**Comment by marina-mosti** [comment]
Gone ahead and implemented the suggestions 👌🏻

**Review by cexbrayat — APPROVED** [review]

**Pull request merged**